sfbrigade/datasci-housing-pipeline
# Incrementally improve data pipeline process

## Description
Cleaner.py  requires a column name file that has key, values pairs in order to process any new csv files. I created a create_column_names.py that dynamically creates header files required to process new files for the cleaner.py file. the create_column_names.py file requires there to be a global header file, which I also created: uniq_header_mappings.txt. 

## Proposed Changes

This pull request will change/create the following things:
For the Cleaner file:
 - 2017 Q2, Q3 raw and cleaned files processed by cleaner and data_model
 - Check for null strings or empty strings in functions in cleaner.py file
 - create new column name mappings based on new file inputs
 - create global unique header mappings for the cleaner.py file

For PostgresDB setup:
- sets up a local postgres db and table
- inserts data from the ../data/cleaned/all_quarters__one_record_per_project.cs file into a table called 'san_francisco_housing_pipeline'


## Testing
To test this pull request, follow these steps:
For the Cleaner file:
1. Download any new quarter csv data from the SF Data Gov website: https://data.sfgov.org/Housing-and-Buildings/SF-Development-Pipeline-2018-Q1/dg6z-zdpi
2. change the file name to match convention "San_Francisco_Development_Pipeline_2018_Quarter_1.csv" and move it to the ./raw directory
3. Open the create_column_names.py file and insert the string "San_Francisco_Development_Pipeline_2018_Quarter_1.csv" into line 67
4. Run the create_column_names.py file
5. Observe that a new .txt. file is created in the column names.
6. Run the cleaner.py file, then the data_model.py file to produce a new analysis.

For PostgresDB setup:
1. Install postgres or check on command line if already installed:
- for mac $psql
- if errors: install psql locally: https://www.postgresql.org/download/macosx/
2. Hop into the setup.py file. Line 145 - 147, set up your user, host and password for the psql db
3. Run the setup.py file or $python setup.py on command line. There should be an output that shows how many rows have been inserted into the table
4. $ psql from the command line to access the table in the db. 
5. Connect to the san_francisco_housing_pipeline
- $\c san_francisco_housing_pipeline;
6. Verify the sum of the rows match up with the output from the setup.py file
- $ select count(*) from san_francisco_housing_pipeline;
- should output: 3551 entires

## Merge Checklist
Each of these boxes should be checked before merge:
- [ ] Approved by at least 1 reviewer
- [ ] Tested with the above procedure
